File: HardwareInfoCollector/models/ram.py
from models.hardware import *


class Ram(Hardware):
    DDR5: bool
    Frequency: int
    Capacity: int
    Count: int

    def __init__(self, product_type: int, model: str, price: int, link: str,
                 ddr5: bool, frequency: int, capacity: int, count: int):
        super().__init__(product_type, model, price, link)
        self.DDR5 = ddr5
        self.Frequency = frequency
        self.Capacity = capacity
        self.Count = count

    def __str__(self):
        return f'{self.ProductType} {self.Model} {self.Price} {self.Link} {self.DDR5} ' \
               f'{self.Frequency} {self.Capacity} {self.Count}'

File: BuildGenerator/src/Bll/Models/Hardware/Ram.cs
namespace Bll.Models.Hardware;

public class Ram : Abstractions.Hardware
{
    public double Price { get; set; }
    public string? Link { get; set; }
    public string? Model { get; set; }

    public bool Ddr5 { get; set; }
    public int Frequency { get; set; }
    public int Capacity { get; set; }
    public int Count { get; set; }

    public Ram()
    {
        
    }
    
    public Ram(double price, string link, string model, bool ddr5, int frequency, int capacity, int count)
    {
        Price = price;
        Link = link;
        Model = model;
        Ddr5 = ddr5;
        Frequency = frequency;
        Capacity = capacity;
        Count = count;
    }
}
